imdown-mobile/events/event_detailed.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from "react";
+import { FlatList, Image, Linking, Text, TouchableOpacity, View } from "react-native";
+import { EventDetailedStyles } from "./event_detailed_styles";
+import moment from 'moment';
+import Divider  from '../components/divider/divider'
+
+const DOWN_EMOJI_HEIGHT = 82
+const DOWN_EMOJI_WIDTH = 85
+const EVENT_PIC_HEIGHT = 130
+const EVENT_PIC_WIDTH = 130
+const ROW_BUTTON_HEIGHT = 40
+const ROW_BUTTON_WIDTH = 40
+
+const EventDetailed = (props) => {
+const [event, setEvent] = useState(props.route.params.event)
+
+	const renderRSVPUser = ({ item }: { item: String }) => {
+    return (
+      <View>
+            <Text style={EventDetailedStyles.rsvp_user}>{item}</Text>
+      </View>
+    );
+	};
+	
+  return (
```

Comment:
I think since this return is getting kind of large, it would be nice to break it down into separate functions. For example, we can move the button row code into a separate function `renderButtonRow` like so:
```
const renderButtonRow = () => {
<View style={EventDetailedStyles.button_row_container}>
            <TouchableOpacity style ={EventDetailedStyles.delete_event_container}>
                <Image source = {require('../assets/delete_icon.png')} style = {{ width: ROW_BUTTON_HEIGHT, height: ROW_BUTTON_WIDTH }}/> 
                <Text style={EventDetailedStyles.button_row_text}> delete </Text>
            </TouchableOpacity>
            <TouchableOpacity style={EventDetailedStyles.decline_rsvp_container}>
                <Image source = {require('../assets/cancel_rsvp.png')} style = {{ width: ROW_BUTTON_HEIGHT, height: ROW_BUTTON_WIDTH }}/> 
                <Text style={EventDetailedStyles.button_row_text}> decline </Text>
            </TouchableOpacity>
            <TouchableOpacity style ={EventDetailedStyles.edit_event_container}>
                <Image source = {require('../assets/edit_event.png')} style = {{ width: ROW_BUTTON_HEIGHT, height: ROW_BUTTON_WIDTH }}/> 
                <Text style={EventDetailedStyles.button_row_text}> edit </Text>
            </TouchableOpacity>
        </View>
}
```

Then the return itself would just contain 
```
return (
...
{renderButtonRow()}
```

Reply:
sure